docs/backendFlow.md

```diff
@@ -0,0 +1,243 @@
+# **Backend Flow 🕸️**
+
+<br>
+
+> ## Explaining the flow using `Users` created in the project ✌️
+
+<br>
+
+- [**Backend Flow 🕸️**](#backend-flow-️)
+  - [Controllers](#controllers)
+  - [Models](#models)
+  - [Base/ Global Route](#base-global-route)
+  - [Different HTTP Routes](#different-http-routes)
+  - [Middlewares](#middlewares)
+  - [config.js](#configjs)
+  - [server.js](#serverjs)
+  - [test](#test)
+  - [package.json file](#packagejson-file)
+    - [Want to Contribute ??](#want-to-contribute-)
+
+## Controllers
+
+- Contains the business logic
+- Here the request and response are handled
+
+👀 Example:
+
+`getUsers Controller`
+
+```JS
+const getUsers = async (req, res) => {
+  try {
+    const allUsers = await userQuery.fetchUsers(req.query)
+    return res.json({
+      message: 'Users returned successfully!',
+      users: allUsers
+    })
+  } catch (error) {
+    logger.error(`Error while fetching all users: ${error}`)
+    return res.boom.serverUnavailable('Something went wrong please contact admin')
+  }
+}
+```
+
+<br>
+
+## Models
+
+- Interacts with the database, Ex. [Firebase](https://firebase.google.com/docs/guides "Firebase Docs")
+- Triggers database calls
+
+👀 Example:
+
+`fetchUsers Model`
+
+```JS
+const fetchUsers = async (query) => {
+  try {
+    const snapshot = await userModel
+      .limit(parseInt(query.size) || 100)
+      .offset((parseInt(query.size) || 100) * (parseInt(query.page) || 0))
+      .get()
+
+    const allUsers = []
+
+    snapshot.forEach((doc) => {
+      allUsers.push({
+        id: doc.id,
+        ...doc.data(),
+        phone: undefined,
+        email: undefined,
+        tokens: undefined
+      })
+    })
+
+    return allUsers
+  } catch (err) {
+    logger.error('Error retrieving user data', err)
+    throw err
+  }
+}
+```
+
+<br>
+
+## Base/ Global Route
+
+- In project, base route `'/users'` means this is the base for every users route
+  [ From: `routes> index.js` ]
+
+👀 Example:
+
+`routes/index.js`
+
+```
+app.use('/users', require('./users.js'))
+```
+
+Description:
+
+- To access details of user with `userId`, the route will be `/users/userId` and so on.
+
```

Comment:
I guess this statement **alone** can confuse people
Can we include an example?
Like:
https://github.com/Real-Dev-Squad/website-backend/blob/b42e3b5866a390fe99de85fa078d060b4f1ab53e/routes/users.js#L211

Reply:
Ok can be done

Reply:
Yes, Can be added